### PR TITLE
Add development toctree and PyGMT team page

### LINKED ISSUE #1212: Adding a PyGMT team page
Background information:

1. It could be helpful for new contributors to know who the PyGMT maintainers are, now that the list of maintainers is growing.
2. I am working on improving the authorship guidelines and acknowledgements structure for GMT (see https://github.com/GenericMappingTools/gmt/issues/2330 for context) and envision one of the products being similar to the [Numpy team gallery](https://numpy.org/gallery/team.html), which could be hosted at www.generic-mapping-tools.org with a link from the [about page](https://www.generic-mapping-tools.org/about/). PyGMT maintainers could be included on that page, as an option.
 
So here are my questions regarding PyGMT specifically:

1. Do people think it would be helpful to publicly list the PyGMT maintainers? 
2. Are maintainers comfortable with being included on some sort of team page? (this could be discussed on slack)
3. If so, should this be done on the same page as the GMT teams or a separate page? If it's a separate page, should it be hosted on www.generic-mapping-tools.org or elsewhere?

*Originally posted in https://github.com/GenericMappingTools/pygmt/issues/1113#issuecomment-805343339*.

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

This PR adds a PyGMT team page under a development toctree in the docs.

Fixes #1212 

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by weiji14 — COMMENTED** [review]
Not sure if it helps, but you can actually write a markdown document instead, since we have the [MyST](https://myst-parser.readthedocs.io/en/latest/index.html) parser. Might be less painful than writing pure HTML or RST.

**Comment by meghanrjones** [comment]
> Not sure if it helps, but you can actually write a markdown document instead, since we have the [MyST](https://myst-parser.readthedocs.io/en/latest/index.html) parser. Might be less painful than writing pure HTML or RST.

Do you think pure HTML will be too hard to maintain? I did not find it too hard to write, but can research other options if necessary. I did not know of a way to create this format in markdown.

**Comment by core-man** [comment]
[Numpy: About us](https://numpy.org/about/) lists the [Numpy team gallery](https://numpy.org/gallery/team.html). See the original markdown files:
- [numpy.org/conent/en/about.md](https://github.com/numpy/numpy.org/blob/master/content/en/about.md)
- [numpy.org/scripts/gallery](https://github.com/numpy/numpy.org/tree/master/scripts/gallery)

It seems that it's a little easier to create a simple gallery than markdown at the beginning.

**Comment by meghanrjones** [comment]
> [Numpy: About us](https://numpy.org/about/) lists the [Numpy: Team gallery](https://numpy.org/gallery/team.html). See the corresponding markdown files:
> 
>     * [numpy.org/conent/en/about.md](https://github.com/numpy/numpy.org/blob/master/content/en/about.md)
> 
>     * [numpy.org/scripts/gallery](https://github.com/numpy/numpy.org/tree/master/scripts/gallery)
> 
> 
> It seems that it's a little easier to create a simple gallery using HTML than markdown at the beginning.

Thanks for sharing this - it's helpful to know how that page was built. I agree that it seems more complicated than raw html for now, since it requires several different files.

**Comment by michaelgrund** [comment]
I really like the HTML version (especially the pics look nice in that large format). On the other side, as the others mentioned, it should be also possible to easily maintain the whole thing.

File: doc/_static/style.css
Comment (seisman) [inline doc/_static/style.css:205]: Can we add some spaces between columns? 

I don't know much about CSS, but the following changes look better to me:
```suggestion
    width: 18%;
    margin: 0.25em;
```

**Current**:

image


**After**
image

Comment (meghanrjones) [reply doc/_static/style.css]: Looks good.

File: doc/team.rst
Comment (weiji14) [inline doc/team.rst:75]: Typo on Leo's surname.
```suggestion
            <h6 class="team-card-text">Leonardo Uieda</h6>
```

File: doc/team.rst
Comment (weiji14) [inline doc/team.rst:9]: Minor tweaks to the text and some word wrapping (not sure if we should wrap to 79 char or 88 char tbh, I usually do 79 char out of habit).

```suggestion
We are an international team dedicated to building a Pythonic API for the
Generic Mapping Tools (GMT). Our goal is to improve GMT's accessibility for
new users, by creating user-friendly interfaces with the GMT C API, supporting
rich display in Jupyter notebooks, and integrating with the PyData ecosystem.
```

**Review by weiji14 — COMMENTED** [review]
I don't have the time to go down the rabbit hole of converting HTML to MyST/Markdown, so let's just stick with HTML.

I did consider all-contributors as an alternative before in https://github.com/GenericMappingTools/gmt/issues/2330#issuecomment-569551125, but it's a bit more complex to configure and the PyGMT development team isn't quite at the size to warrant that yet.

File: doc/team.rst
Comment (weiji14) [inline doc/team.rst:67]: [Emeritus](https://en.wikipedia.org/wiki/Emeritus) sounds a bit old for these two young fellas, is there a better term maybe?

File: doc/team.rst
Comment (weiji14) [inline doc/team.rst:13]: 'Maintainers' is correct, but I'd prefer a term like "Core team" or similar so that people won't feel like joining the team means maintaining software only (which is part of our job, but not the whole of it). Any other suggestions?

Comment (core-man) [reply doc/team.rst:13]: **Core Developer**?

Comment (meghanrjones) [reply doc/team.rst:13]: Yes, I agree with your point. I think that specifying who leads project maintenance could be helpful transparency for new contributors, so it would be nice to get a solution that recognizes both maintenance, development, and the rest. Do you think it could be an option to have a 'Distinguished Contributors' section in addition to the 'Maintainers' section, with overlap between the two as appropriate? I like Jupyter's [distinguished contributors](https://jupyter.org/about) section because the path for recognition is clearly described. Or we could do a 'Core Developers' team like @core-man suggested, with some tag for people who are also maintainers. But, it is possible that someone may be interested in being a maintainer without wanting to spend as much time on development.

File: doc/team.rst
Comment (meghanrjones) [reply doc/team.rst:67]: Oops, my bad. Maybe 'Former Maintainers'? Or we could combine the sections with some design choice to indicate activate maintainers.

File: doc/team.rst
Comment (meghanrjones) [reply doc/team.rst:13]: Just want to follow up on this suggestion @weiji14 and @core-man, do you prefer a single "core team" to having two categories, one for active maintainers and one for core developers/core contributors/distinguished contributors? My preference is the second option, but I am open to either.

Comment (core-man) [reply doc/team.rst:13]: > I think that specifying who leads project maintenance could be helpful transparency for new contributors, so it would be nice to get a solution that recognizes both maintenance, development, and the rest.

Sounds good. This could help new contributors to have a quick recognition of the PyGMT Team. But the current core developers are all maintainers, right?

> do you prefer a single "core team" to having two categories, one for active maintainers and one for core developers/core contributors/distinguished contributors?

If we only create one team called **core team**, do we have to create a Level-2 title for it (e.g., **Maintainers** and **Emeritus Maintainers** are Level-2 titles now)? Currently, the level-1 title is called **PyGMT Team Gallery**.

We have a few categories and they could be active or not active. I am a little confused about them.
- **Maintainers**
- **Core Developers**
- **Core Contributors**
- **Distinguished Contributors**
- **Contributors**

I think all of you belong to **Core Developers**, and all of you are also maintainers, right? So, I don't think we need to distinguish **Core Developers** and **Maintainers** unless we have one core developer who is not a maintainer, vice versa.

Could we add a labeler, **active**, for those core developers/maintainers who are active now? Thus, it does not need to create a separate category for core developers/maintainers who are not active now.

> My preference is the second option, but I am open to either.

:smile: Sorry that I'm not clear what the first and second options are.

Comment (meghanrjones) [reply doc/team.rst:13]: OK, I will make a revision accordingly and then you can check it out to let me know if I understood your comments correctly.

Comment (meghanrjones) [reply doc/team.rst]: I pushed a revised version based on my interpretation of your comments. I think we need a statement at the top that invites people to get involved and (ideally) lays out the criteria for becoming part of the 'core team' (if this iteration stays). To be honest, while I have spent enough time on PyGMT maintenance to have been comfortable with the maintainers group, it feels odd to me to be the one including myself and listing others on the 'core team' group when I have only been involved with PyGMT for a small amount of time compared to, for example, Wei Ji and Dongdong. I would rather it be better defined somewhere how the 'core team' gets decided and adjusted over time (maybe that guidance exists already, I am not sure).

Comment (weiji14) [reply doc/team.rst]: > To be honest, while I have spent enough time on PyGMT maintenance to have been comfortable with the maintainers group, it feels odd to me to be the one including myself and listing others on the 'core team' group when I have only been involved with PyGMT for a small amount of time

It is not the amount of time you've spent, but rather, what you can do that counts :)

At this stage, PyGMT is still a fairly small team (<10 people), and @core-man raises a good point that there's not much of a distinction between developers or maintainers per se. I'd be tempted to throw @core-man into the PyGMT team gallery list since he's been very helpful with a lot of the documentation and design/decision making.

> Could we add a labeler, active, for those core developers/maintainers who are active now? Thus, it does not need to create a separate category for core developers/maintainers who are not active now. We only have two non-active core developers now.

What about an asterisk (*) after the name to indicate that a person is active? Or have the name be **bold** when someone is active?

Comment (michaelgrund) [reply doc/team.rst]: > At this stage, PyGMT is still a fairly small team (<10 people), and @core-man raises a good point that there's not much of a distinction between developers or maintainers per se. I'd be tempted to throw @core-man into the PyGMT team gallery list since he's been very helpful with a lot of the documentation and design/decision making.
> 
Fully agree!

> > Could we add a labeler, active, for those core developers/maintainers who are active now? Thus, it does not need to create a separate category for core developers/maintainers who are not active now. We only have two non-active core developers now.
> 
> What about an asterisk (*) after the name to indicate that a person is active? Or have the name be **bold** when someone is active?

If possible, I would prefer somekind of label as suggested by @core-man.

Comment (core-man) [reply doc/team.rst]: I like the current design, and I think any way above to indicate active maintainers/developers are okay. When we have a way to indicate active maintainers/developers soon, new contributors could easily recognize the whole team due to the limited number.

BTW, I feel new contributors could also know the details about PyGMT Teams by [pygmt-team](https://github.com/orgs/GenericMappingTools/teams/pygmt-team/teams), which has a very good and clear layout.

> I'd be tempted to throw @core-man into the PyGMT team gallery list since he's been very helpful with a lot of the documentation and design/decision making.

It feels odd to me to be included in the PyGMT team gallery list as compared to the big amount of work and time you spent. Meanwhile, I am afraid that I will still have limited time to help with only a few parts of PyGMT (i.e., tutorials/gallery/documentation review, and some module enhancement/wrapping interesting to me (e.g., `plot`/`plot3d`/`sac`/`wiggle`/`histogram` currently))

Comment (weiji14) [reply doc/team.rst]: Ok, I see that people are keen on having an **active** label/tag somehow. Will leave it up to @meghanrjones on how to include this in the team gallery and make it look nice :slightly_smiling_face:

> > I'd be tempted to throw @core-man into the PyGMT team gallery list since he's been very helpful with a lot of the documentation and design/decision making.

> It feels odd to me to be included in the PyGMT team gallery list as compared to the big amount of work and time you spent. Meanwhile, I am afraid that I will still have limited time to help with only a few parts of PyGMT (i.e., tutorials/gallery/documentation review, and some module enhancement/wrapping interesting to me (e.g., plot/plot3d/sac/wiggle/histogram currently))

Trust me @core-man, you've probably done more in the past three months in terms of features/documentation/review work than what I did in my first year with PyGMT :laughing:. And it's ok to focus on the parts you use such as the seismological modules. I would encourage you to be on the team list just to show that PyGMT values new contributors that can do good work and isn't restricted to an exclusive club of developers, but I respect your decision if you don't feel ready to be added yet.

Comment (meghanrjones) [reply doc/team.rst]: https://github.com/GenericMappingTools/pygmt/pull/1308/commits/9dbf864ef42873e83467ddbd47c516b7831f23fb added an 'active' label.

https://github.com/GenericMappingTools/pygmt/pull/1308/commits/52d814123ca516c1aac9094c13534e8972846237 added @core-man. @core-man, I will respect your preferences for whether you want to accept/reject inclusion on the page, so feel free to let me know. I agree with all the previous comments that your contributions are really valuable.

https://github.com/GenericMappingTools/pygmt/pull/1308/commits/d302e67baea0ac5f1d46fe995de932b9d38307dc restructured the page following Jupyter's Distinguished Contributors model, with a label of 'Maintainer' for active maintainers. I prefer this because it meets the two main goals of 1. acknowledging people who have put a lot into the project without requiring any responsibilities from them and 2. providing helpful transparency about who is responsible for the current maintenance of the project. I also prefer having it stated when/how it gets updated. Easy to revert if others do not like it.

**Comment by weiji14** [comment]
Oops, sorry @meghanrjones!! I was cleaning up some local git branches and accidentally deleted this branch :frowning:

File: doc/team.rst
Comment (core-man) [reply doc/team.rst]: Thanks a lot for including me in the team gallery. I will continue to do what I can do for the project.

File: doc/team.rst
Comment (core-man) [inline doc/team.rst:18]: See [gmt/AUTHORS.md](https://github.com/GenericMappingTools/gmt/blob/master/AUTHORS.md):
> ## The Founders
> GMT was conceived and designed in the late 1980s and early 1990s by Paul Wessel and Walter H.F. Smith. The > two founders built and maintained GMT from version 1 through 3.

I have another thought. How about adding a short description (or a single category)  for PyGMT founders? That's a great way to show respect to them. I know **Leonardo** is the first to start the PyGMT project, and **Dongdong** and **weiji14** soon joined the project and continue to contribute to the project. We could have more teams and more active contributors in the future, and re-organize the team gallery, but the founders won't change.

Comment (weiji14) [reply doc/team.rst:18]: Leo is mentioned as the founder at https://www.pygmt.org/dev/overview.html. I'm not really a founder, more like an early adopter. We could put a 'Founder' label/tag under Leo's name I suppose.

Comment (meghanrjones) [reply doc/team.rst:18]: OK, so then I should probably also add Paul with a 'Founder' tag, since I think him and Leo were both involved with getting the funding to start the project which is a substantial contribution. Sound OK?

Comment (core-man) [reply doc/team.rst:18]: > OK, so then I should probably also add Paul with a 'Founder' tag, since I think him and Leo were both involved with getting the funding to start the project which is a substantial contribution. Sound OK?

Sound good. But it's better to have more comments/thoughts from @weiji14 and @seisman since they might know more about the history.

Comment (weiji14) [reply doc/team.rst:18]: Most of what I know about PyGMT's early history (or gmt-python as it was back then) is from Leo's Scipy 2017 talk (https://www.youtube.com/watch?v=93M4How7R24). I'm ok with adding Paul in.

Comment (meghanrjones) [reply doc/team.rst:18]: Done in https://github.com/GenericMappingTools/pygmt/pull/1308/commits/d0e22aa1370c35cd142910cca1aa3443f8aa7813

File: doc/team.rst
Comment (core-man) [inline doc/team.rst:15]: On my mac (13 inch), the display of **Leonardo** is not perfect for 100% view, while 90% is okay. Meanwhile, my workstation is okay since it has a monitor of 23 inch.

100% | 90%
--- | ---
 |

Comment (meghanrjones) [reply doc/team.rst:15]: Thanks for pointing this out. I updated the css style, but unfortunately I cannot figure out how to make the items both a nice grid and also flexible to fit the contents. Please let me know if you think the new style is better.

File: doc/team.rst
Comment (seisman) [inline doc/team.rst:4]: I think "Team Gallery" is enough.
```suggestion
Team Gallery
============
```

**Comment by seisman** [comment]
It looks pretty bad on my Microsoft Edge and Safari on macOS:

image

File: doc/team.rst
Comment (seisman) [inline doc/team.rst:22]: > selected bi-annually by those listed below.

I might miss some discussions, but have you reached an agreement about the frequency? Do you mean "once every two years" or "twice per year"? As a non-native English speaker, it sounds like "once every two years", which is too long for an active project like PyGMT.

When I search for "bi-annually", I found this (https://www.merriam-webster.com/dictionary/biannual):

> When we describe something as biannual, we can mean either that it occurs twice a year or that it occurs once every two years. So how does someone know which particular meaning we have in mind? Well, unless we provide them with a contextual clue, they don't.

File: doc/team.rst
Comment (core-man) [reply doc/team.rst:15]: I have the same issue as **Dongdong** pointed in https://github.com/GenericMappingTools/pygmt/pull/1308#issuecomment-867300192 now.

**Comment by michaelgrund** [comment]
> 
> 
> It looks pretty bad on my Microsoft Edge and Safari on macOS:
> 
What do you mean with bad here? The arrangements of the indiviudal items? Using Firefox on my Windows machine it looks similar but the items are arranged in a different way, e.g. seisman is not "alone" in the top line 😉 

team_gallery_firefox

**Comment by seisman** [comment]
> What do you mean with bad here @seisman ?

What I saw is two columns per row and each item has a huge size.

File: doc/team.rst
Comment (meghanrjones) [reply doc/team.rst]: Thanks for pointing out the confusing language. I meant twice per year and updated to specify that. It's open for discussion, I just selected 2x per year as an arbitrary starting point.

**Comment by meghanrjones** [comment]
> > What do you mean with bad here @seisman ?
> 
> What I saw is two columns per row and each item has a huge size.

OK I updated and tested on different browsers using macOS.

**Review by weiji14 — APPROVED** [review]
Looks good to me (on Firefox and Brave browser in Linux). But wait for the others to confirm that it looks ok on their browsers.

**Review by core-man — APPROVED** [review]
Good to me, too. I use Chrome and Firefox on CentOS7.5 (22 inch), and Safari and Chrome on macOS (13 inch)

**Review by michaelgrund — APPROVED** [review]
Looks also good to me using Firefox on Windows 10.

**Comment by meghanrjones** [comment]
Anyone have ideas for why the style checks are failing? I did not touch those files, and the checks pass locally.

**Comment by seisman** [comment]
> Anyone have ideas for why the style checks are failing? I did not touch those files, and the checks pass locally.

pylint just released v2.9.1 a few hours ago (https://github.com/PyCQA/pylint/releases/tag/v2.9.1). The pylint errors are new and should be fixed in a separate PR.

**Comment by meghanrjones** [comment]
> > Anyone have ideas for why the style checks are failing? I did not touch those files, and the checks pass locally.
> 
> pylint just released v2.9.1 a few hours ago (https://github.com/PyCQA/pylint/releases/tag/v2.9.1). The pylint errors are new and should be fixed in a separate PR.

thanks, sounds good.

File: doc/team.rst
Comment (seisman) [inline doc/team.rst:12]: Is it a typo?
```suggestion
All are welcome to become involved with the PyGMT project! For more information
```

**Review by seisman — COMMENTED** [review]
The new design looks good to me. Besides a possible typo, should we use markdown instead of ReStructureText for this file, since we now use the [MyST](https://myst-parser.readthedocs.io/en/latest/) parser and other files (contributing, maintaining) are also markdown files?

**Comment by meghanrjones** [comment]
> The new design looks good to me. Besides a possible typo, should we use markdown instead of ReStructureText for this file, since we now use the [MyST](https://myst-parser.readthedocs.io/en/latest/) parser and some other files (contributing, maintaining) are also markdown files?

I changed to markdown, even though I don't really understand the benefits of using MyST over traditional ReST with sphinx. Easy to revert if people don't like this change.

**Comment by core-man** [comment]
> Besides a possible typo, should we use markdown instead of ReStructureText for this file, since we now use the [MyST](https://myst-parser.readthedocs.io/en/latest/) parser and some other files (contributing, maintaining) are also markdown files?

I feel it's okay to use ReST. Instead, shall we use ReSR for contributing and maintaining? We used Makrdown since they are original put in the repository.

**Comment by seisman** [comment]
> > The new design looks good to me. Besides a possible typo, should we use markdown instead of ReStructureText for this file, since we now use the [MyST](https://myst-parser.readthedocs.io/en/latest/) parser and some other files (contributing, maintaining) are also markdown files?
> 
> I changed to markdown, even though I don't really understand the benefits of using MyST over traditional ReST with sphinx. Easy to revert if people don't like this change.

See https://github.com/GenericMappingTools/pygmt/pull/941.

The main reason to use the MyST is to avoid the pain of having to convert the markdown changelog generated by Release Drafter into ReST syntax (https://github.com/GenericMappingTools/pygmt/issues/885#issue-807847528).

For other files (contributing or maintaining), using ReST or Markdown almost makes no difference, but I feel new contributors may be more familiar with Markdown syntax.

**Comment by meghanrjones** [comment]
> > > The new design looks good to me. Besides a possible typo, should we use markdown instead of ReStructureText for this file, since we now use the [MyST](https://myst-parser.readthedocs.io/en/latest/) parser and some other files (contributing, maintaining) are also markdown files?
> > 
> > 
> > I changed to markdown, even though I don't really understand the benefits of using MyST over traditional ReST with sphinx. Easy to revert if people don't like this change.
> 
> See #941.
> 
> The main reason to use the MyST is to avoid the pain of having to convert the markdown changelog generated by Release Drafter into ReST syntax ([#885 (comment)](https://github.com/GenericMappingTools/pygmt/issues/885#issue-807847528)).
> 
> For other files (contributing or maintaining), using ReST or Markdown almost makes no difference, but I feel new contributors may be more familiar with Markdown syntax.

OK, I will leave as markdown and will merge in tomorrow unless anyone wants to veto with a thumbs down.

**Review by seisman — APPROVED** [review]

**Pull request merged**
